### PR TITLE
Upgrade to Ruby 2.7

### LINKED ISSUE #2035: Upgrade Ruby version
Sqreen triggered an alert about the Ruby runtime version ([2.6.6](https://github.com/DCAFEngineering/dcaf_case_management/blob/main/Gemfile#L2)): 

> Vulnerable or outdated runtime in use

The current normal ruby maintenance version is 2.7, so we should probably upgrade to that.
https://www.ruby-lang.org/en/downloads/branches/

### PR DESCRIPTION
**Pull request by lomky**
I rule and have completed some work on Case Manager that's ready for review!

Migrating us up to Ruby 2.7

This pull request makes the following changes:
* Gemfile bump
* Circular dependency bugfix
* CircleCI & Rubocop
* Docker Image

It relates to the following issue #s: 
* Fixes #2035

### CODE REVIEW

**Comment by xmunoz** [comment]
I did a fresh checkout and followed the instructions in `SETUP.md`. There was one serious error and one nit.

Error:
```
$ docker-compose build
...
[3/4] Linking dependencies...
warning " > bootstrap@4.4.1" has unmet peer dependency "popper.js@^1.16.0".
warning " > webpack-dev-server@3.10.3" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning "webpack-dev-server > webpack-dev-middleware@3.7.2" has unmet peer dependency "webpack@^4.0.0".
[4/4] Building fresh packages...
...
```

Nit:

```
$ docker-compose run --rm web rake db:seed
Starting dcaf_case_management_db_1 ... done
warning: parser/current is loading parser/ruby27, which recognizes
warning: 2.7.1-compliant syntax, but you are running 2.7.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
...
```

When I tried to load the app in my browser, the app error-ed out.
```
web_1        | [Webpacker] [ERROR] Compilation failed:
web_1        | yarn run v1.22.5
web_1        | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
web_1        | 
web_1        | 
web_1        | error Command "webpack" not found.
web_1        | 
web_1        | [INFO ] Completed 500 Internal Server Error in 9969ms (MongoDB: 0.0ms | Allocations: 612577)
web_1        | [DEBUG] Webpacker can't find application.css in /usr/src/app/public/packs/manifest.json. Possible causes:
web_1        | 1. You want to set webpacker.yml value of compile to true for your environment
web_1        |    unless you are using the `webpack -w` or the webpack-dev-server.
web_1        | 2. webpack has not yet re-run to reflect updates.
web_1        | 3. You have misconfigured Webpacker's config/webpacker.yml file.
web_1        | 4. Your webpack configuration is not creating a manifest.
web_1        | Your manifest contains:
web_1        | {
web_1        | }
web_1        |  excluded from capture: DSN not set
web_1        | [FATAL]   
web_1        | ActionView::Template::Error (Webpacker can't find application.css in /usr/src/app/public/packs/manifest.json. Possible causes:
web_1        | 1. You want to set webpacker.yml value of compile to true for your environment
web_1        |    unless you are using the `webpack -w` or the webpack-dev-server.
web_1        | 2. webpack has not yet re-run to reflect updates.
web_1        | 3. You have misconfigured Webpacker's config/webpacker.yml file.
web_1        | 4. Your webpack configuration is not creating a manifest.
web_1        | Your manifest contains:
web_1        | {
web_1        | }
web_1        | ):
web_1        |      9:     <%= csp_meta_tag %>
web_1        |     10:     <%= favicon_link_tag %>
web_1        |     11: 
web_1        |     12:     <%= stylesheet_pack_tag 'application', 'data-search-pseudo-elements': true, 'data-turbolinks-track': 'reload' %>
web_1        |     13:     <%= stylesheet_link_tag :application, media: 'all' %>
web_1        |     14: 
web_1        |     15:     <%= javascript_pack_tag 'application', 'data-search-pseudo-elements': true, 'data-turbolinks-track': 'reload' %>
web_1        |   
web_1        | app/views/layouts/application.html.erb:12
```

**Review by xmunoz — CHANGES_REQUESTED** [review]
Couldn't get the app running locally with these changes. See above.

**Comment by xmunoz** [comment]
Here is what I saw in the browser
Screenshot from 2020-11-11 14-56-32

**Comment by xmunoz** [comment]
Another possible hint: when I ran `docker-compose up`, webpacker spits this out:

```
webpacker_1  | error An unexpected error occurred: "https://registry.yarnpkg.com/webpack-dev-server: unable to get local issuer certificate".
```

**Comment by lomky** [comment]
@xmunoz I added some resolutions for that initial issue in the build, and I believe that fixed the error in the webpack step later. I'm not too concerned about the nit, but thanks for raising it! Give it a shot now?

**Comment by xmunoz** [comment]
The good news is that I can now load the app in the web browser and log in! The bad news is that console is still spitting our errors:
```
web_1        | [8] * Listening on tcp://0.0.0.0:3000
web_1        | [8] Use Ctrl-C to stop
web_1        | [8] - Worker 0 (pid: 22) booted, phase: 0
web_1        | [8] - Worker 1 (pid: 34) booted, phase: 0
webpacker_1  | info fsevents@2.1.3: The platform "linux" is incompatible with this module.
webpacker_1  | info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
webpacker_1  | info fsevents@1.2.7: The platform "linux" is incompatible with this module.
webpacker_1  | info "fsevents@1.2.7" is an optional dependency and failed compatibility check. Excluding it from installation.
webpacker_1  | [3/4] Linking dependencies...
webpacker_1  | [4/4] Building fresh packages...
webpacker_1  | Done in 16.96s.
webpacker_1  | /usr/local/bundle/gems/bundler-2.1.4/lib/bundler/definition.rb:495:in `validate_ruby!': Your Ruby version is 2.6.6, but your Gemfile specified 2.7.2 (Bundler::RubyVersionMismatch)
webpacker_1  | 	from /usr/local/bundle/gems/bundler-2.1.4/lib/bundler/definition.rb:470:in `validate_runtime!'
webpacker_1  | 	from /usr/local/bundle/gems/bundler-2.1.4/lib/bundler.rb:143:in `setup'
webpacker_1  | 	from /usr/local/bundle/gems/bundler-2.1.4/lib/bundler/setup.rb:20:in `block in <top (required)>'
webpacker_1  | 	from /usr/local/bundle/gems/bundler-2.1.4/lib/bundler/ui/shell.rb:136:in `with_level'
webpacker_1  | 	from /usr/local/bundle/gems/bundler-2.1.4/lib/bundler/ui/shell.rb:88:in `silence'
webpacker_1  | 	from /usr/local/bundle/gems/bundler-2.1.4/lib/bundler/setup.rb:20:in `<top (required)>'
webpacker_1  | 	from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
webpacker_1  | 	from /usr/local/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
webpacker_1  | 	from bin/webpack-dev-server:10:in `<main>'
dcaf_case_management_webpacker_1 exited with code 1
```

The error appears to be originating from [this image](https://hub.docker.com/layers/colinxfleming/dcaf_case_management/latest/images/sha256-a0bc4ba93af0ec9f18e69239550b19917df0bc62dfdcec478c808a879656f77b?context=explore), which was built with [this Dockerfile](https://github.com/DCAFEngineering/dcaf_case_management/blob/main/.docker/Dockerfile). Do you want to update, build and push an upgraded version of this image as part of this PR, or do that as a follow-up task?

File: app/helpers/dashboards_helper.rb [hunk 1]
```diff
@@ -8,15 +8,15 @@ def date_range(date: Time.zone.now)
     end
   end
 
-  def month_range(date: date)
+  def month_range(date: Time.zone.now)
```
Comment (mdworken) [inline app/helpers/dashboards_helper.rb:11]: Is this change intentional?

**Review by mdworken — DISMISSED** [review]
Left a question because I'm not sure if this change was intentional? The rest looks reasonable in that versions went either up within a minor version or went to more modern versions that were not missing security updates as far as I can tell.

File: app/helpers/dashboards_helper.rb (hunk 1 above)
Comment (lomky) [reply app/helpers/dashboards_helper.rb:11]: This was an existing bug that became breaking in the 2.7 upgrade!

Comment (mdworken) [reply app/helpers/dashboards_helper.rb:11]: Damn kwargs 🤣 
image

**Comment by colinxfleming** [comment]
Taking this for a spin!

in rbenv:
* can bundle install
* can rails test
* can spin up localhost and log in

from docker:
* can docker-compose build
* can docker-compose run web rails test
* I also experience @xmunoz 's issue around the webpacker image pulling from remote instead of the locally built image. I'm going to propose a commit to master that should square this away (and eliminate the dependency on dockerhub 😬  sorry everyone). Implementing this doesn't load it for me but tbh I've always had problems loading docker on this machine, and at the very least it doesn't throw a ton of ruby version errors in the webpack container.

notes:
* I made the serious life error of upgrading to big sur and it made a series of gems (puma, bson_ext, sq_mini_racer) fail. I think all of these error might be solvable by gem upgrades; I got past puma and bson_ext by passing install flags, `gem install bson_ext -v '1.5.1' --source 'https://rubygems.org/' -- --with-cflags="-Wno-error=implicit-function-declaration"` in the case of bson_ext and the same --with-cflags in the case of puma. sq_mini_racer we should probably upgrade (prob by upgrading the sqreen gem). Not necessary to sort out for this PR but something I'll look at this week.

**Review by colinxfleming — APPROVED** [review]
all good from me!

**Comment by xmunoz** [comment]
It did not work for me. `docker build` completed successfully, but the next step did not:
```
$ docker-compose run --rm web rake db:seed
Creating network "dcaf_case_management_default" with the default driver
ERROR: could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network
```

**Review by xmunoz — APPROVED** [review]
LGTM!
